### PR TITLE
fix: add support packages without author's name

### PR DESCRIPTION
**Pull request by joshuapassos**
In projects without author's name i had error:

```
if (author.email) { author = author + ' ' + author.email; }
                                   ^
TypeError: Cannot read property 'email' of undefined
```

With this fix i change to try add name and after add email and after add an empty string.

Example package with this problem:
https://github.com/react-hook-form/react-hook-form

### CODE REVIEW

File: lib/getPackageReportData.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ function getPackageReportData(packageEntry, callback) {
 
 		var author = json.author
 		if (_.isObject(author)) {
-			author = author.name;
+			author = author.name || author.email || ""
```
Comment (BePo65) [inline lib/getPackageReportData.js]: why do you add ` || author.email` as `author.email` is handled in the next line already?
Wouldn't the following line not be enough?
```suggestion
			author = author.name || ""
```

File: lib/getPackageReportData.js [hunk 2]
```diff
@@ -53,7 +53,7 @@ function getPackageReportData(packageEntry, callback) {
...
 			if (author.email) { author = author + ' ' + author.email; }
```
Comment (BePo65) [inline lib/getPackageReportData.js]: And then we have the next problem here: if `author.name` doesn't exist, the we do not have to add the space character (` + ' '`).
So perhaps we should change the line to something like:
```suggestion
			if (author.email) { 
			  if (author.length > 0) {
			    author += ' ';
			  }
			  author += author.email;
			}
```

File: lib/getPackageReportData.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ function getPackageReportData(packageEntry, callback) {
...
 			if (author.url) { author = author + ' ' + author.url; }
```
Comment (BePo65) [inline lib/getPackageReportData.js]: And the same goes here.

**Review by BePo65 — CHANGES_REQUESTED** [review]
If you agree with my ideas, would you please be so kind to update your pr?

**Comment by joshuapassos** [comment]
Done

File: lib/getPackageReportData.js [hunk 4]
```diff
@@ -53,9 +53,18 @@ function getPackageReportData(packageEntry, callback) {
 
 		var author = json.author
 		if (_.isObject(author)) {
-			author = author.name;
-			if (author.email) { author = author + ' ' + author.email; }
-			if (author.url) { author = author + ' ' + author.url; }
+			author = author.name || ""
```
Comment (BePo65) [inline lib/getPackageReportData.js]: please use single ticks for strings to be consistent

File: lib/getPackageReportData.js [hunk 5]
```diff
@@ -53,9 +53,18 @@ function getPackageReportData(packageEntry, callback) {
...
+
+			if (author.email) {
+			  if (author.length > 0) {
+			    author += ' ';
+			  }
+			  author += author.email;
+			}
+
+			if (author.url) {
+			  author += "" + author.url;
```
Comment (BePo65) [inline lib/getPackageReportData.js]: 1. please use single ticks for strings to be consistent
2. if author.name and / or author.email exist, then we need a space character. If neither exist then author = author,url.

So I think this line should look like line 59...62 or did I miss the point?

**Review by BePo65 — CHANGES_REQUESTED** [review]
Sorry that my last comments did miss some points - so I have some more of them :-)

Besides that: shouldn't we add a few test cases for this problem to 'test/getPackageReportData.test.js'? Possible tests could be

1. test with `author: 'joshuapassos'` (or any other name :-)
2. test with `author: {name: 'joshuapassos'}`
3. test with `author: {email: 'jp@ddd.com'}`
4. test with `author: {email: 'jp@ddd.com', url: 'https://jp.ddd.com'}`

or what combinations of parameters you think are useful.

File: lib/getPackageReportData.js (hunk 4 above)
Comment (joshuapassos) [reply lib/getPackageReportData.js]: Ahh sorry.

File: lib/getPackageReportData.js (hunk 5 above)
Comment (joshuapassos) [reply lib/getPackageReportData.js]: Make sense.

**Comment by BePo65** [comment]
So now this looks really good - I do appreciate your work done.

Now what remains are just a few tests, so that we can detect regression caused by future code modifications ( I know, sometimes I am never satisfied :-)
If you would be kind enough to add some, for instance:
1. test with `author: 'joshuapassos'` (or any other name :-)
2. test with `author: {name: 'joshuapassos'}`
3. test with `author: {email: 'jp@ddd.com'}`
4. test with `author: {email: 'jp@ddd.com', url: 'https://jp.ddd.com'}`

or what combinations of parameters you think are useful.Then we can merge this pr.

**Review by BePo65 — APPROVED** [review]
great :-)

**Comment by kessler** [comment]
LGTM,
@joshuapassos thank you very much for the PR !! before we can merge it could you please follow the instructions here: 
https://github.com/ironSource/opensource-contributor-license-agreement

Thanks!

Kessler

**Comment by kessler** [comment]
@joshuapassos thanks! I just need to fix the end to end test. It's breaking every time we update or change dependencies (we need to find a permanent solution to that) a day or two and I'll merge / publish to npm

**Comment by BePo65** [comment]
@kessler any help needed with refactoring the e2e tests?

**Comment by kessler** [comment]
Yes I'm completely swamped. any help will be appreciated.

**Comment by BePo65** [comment]
@kessler made a draft of the updated e2e test in branch `bepo65/using_package_lock`, but I'm afraid we have to think about the points in discussion #44

**Comment by kessler** [comment]
@BePo65 I hear you but maybe we can merge this thing and fix the other problem later on? I'm just too swamped right now :-(

**Comment by BePo65** [comment]
Sorry, I didn't want to bother you :-)

Then let me fix the remoteVersion thing in e2e first. I will get the data from the server in the fixup and use the data from the package.json as "installedVersion".
This will enable e2e even when the server has newer versions available,
And when the time comes, we can discuss the "installedVersion"  thing.

And good luck with your "other business" ;-))

**Pull request merged**